test/Microsoft.AspNetCore.Sockets.Client.Tests/ConnectionTests.cs

```diff
@@ -0,0 +1,167 @@
+                Assert.Equal(new byte[] { 1, 1, 2, 3, 5, 8 }, sendTcs.Task.Result);
+            }
+        }
+
+        [Fact]
+        public async Task CanReceiveData()
+        {
+            var mockHttpHandler = new Mock<HttpMessageHandler>();
+            mockHttpHandler.Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                .Returns<HttpRequestMessage, CancellationToken>(async (request, cancellationToken) =>
+                {
+                    await Task.Yield();
+
+                    var content = string.Empty;
+                    if (request.RequestUri.AbsolutePath.EndsWith("/poll"))
+                    {
+                        content = "42";
+                    }
+                    return new HttpResponseMessage(HttpStatusCode.OK) { Content = new StringContent(content) };
+                });
+
+            using (var httpClient = new HttpClient(mockHttpHandler.Object))
+            using (var longPollingTransport = new LongPollingTransport(httpClient, new LoggerFactory()))
+            using (var connection = await Connection.ConnectAsync(new Uri("http://fakeuri.org/"), longPollingTransport, httpClient))
+            {
+                Assert.False(connection.Input.Completion.IsCompleted);
+
+                await connection.Input.WaitToReadAsync();
+                Message message;
+                connection.Input.TryRead(out message);
+                using (message)
+                {
+                    Assert.Equal("42", Encoding.UTF8.GetString(message.Payload.Buffer.ToArray(), 0, message.Payload.Buffer.Length));
+                }
+            }
+        }
+
+        [Fact]
+        public async Task CanCloseConnection()
+        {
+            var mockHttpHandler = new Mock<HttpMessageHandler>();
+            mockHttpHandler.Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                .Returns<HttpRequestMessage, CancellationToken>(async (request, cancellationToken) =>
+                {
+                    await Task.Yield();
+                    return new HttpResponseMessage(HttpStatusCode.OK) { Content = new StringContent(string.Empty) };
+                });
+
+            using (var httpClient = new HttpClient(mockHttpHandler.Object))
+            using (var longPollingTransport = new LongPollingTransport(httpClient, new LoggerFactory()))
+            using (var connection = await Connection.ConnectAsync(new Uri("http://fakeuri.org/"), longPollingTransport, httpClient))
+            {
+                Assert.False(connection.Input.Completion.IsCompleted);
+                connection.Output.TryComplete();
+
+                var completionTask = Task.WhenAny(Task.Delay(1000), connection.Input.Completion);
```

Comment:
Notice we don't await the task here but below, after we have drained the channel. Perhaps I should call it `whenAny` task and not the `completionTask`.